pyro/distributions/relaxed_bernoulli.py

```diff
@@ -0,0 +1,133 @@
+            self.ps = self.ps.expand(batch_size, self.ps.size(0))
+            self.logits = self.logits.expand(batch_size, self.logits.size(0))
+            if log_pdf_mask is not None and log_pdf_mask.dim() == 1:
+                self.log_pdf_mask = log_pdf_mask.expand(
+                    batch_size, log_pdf_mask.size(0))
+        super(LogitRelaxedBernoulli, self).__init__(*args, **kwargs)
+
+    def batch_shape(self, x=None):
+        """
+        Ref: :py:meth:`pyro.distributions.distribution.Distribution.batch_shape`.
+        """
+        event_dim = 1
+        ps = self.ps
+        if x is not None:
+            if x.size()[-event_dim] != ps.size()[-event_dim]:
+                raise ValueError("The event size for the data and distribution parameters must match.\n"
+                                 "Expected x.size()[-1] == self.ps.size()[-1], but got {} vs {}".format(
+                                    x.size(-1), ps.size(-1)))
+            try:
+                ps = self.ps.expand_as(x)
+            except RuntimeError as e:
+                raise ValueError("Parameter `ps` with shape {} is not broadcastable to "
+                                 "the data shape {}. \nError: {}".format(ps.size(), x.size(), str(e)))
+
+        return ps.size()[:-event_dim]
+
+    def event_shape(self):
+        """
+        Ref: :py:meth:`pyro.distributions.distribution.Distribution.event_shape`.
+        """
+        event_dim = 1
+        return self.ps.size()[-event_dim:]
+
+    def sample(self):
+        """
+        Ref: :py:meth:`pyro.distributions.distribution.Distribution.sample`.
+        """
+        uniforms = Variable(torch.zeros(self.logits.data.size()).uniform_())
+        eps = _get_clamping_buffer(uniforms)
+        uniforms = uniforms.clamp(min=eps, max=1 - eps)
+        logistic = uniforms.log() - (-uniforms).log1p()
+        z = (logistic + self.logits) / self.temperature
+        return z if self.reparameterized else z.detach()
+
+    def analytic_mean(self):
+        """
+        Ref: :py:meth:`pyro.distributions.distribution.Distribution.analytic_mean`
+        """
+        return self.logits / self.temperature
+
+    def batch_log_pdf(self, x):
+        """
+        Ref: :py:meth:`pyro.distributions.distribution.Distribution.batch_log_pdf`
+        """
+        batch_log_pdf_shape = self.batch_shape(x) + (1,)
+        u = -self.temperature * x + self.logits
+        log_prob = self.temperature.log() + u - 2 * torch.log(1 + torch.exp(u))
+        # XXX this allows for the user to mask out certain parts of the score, for example
+        # when the data is a ragged tensor. also useful for KL annealing. this entire logic
+        # will likely be done in a better/cleaner way in the future
+        if self.log_pdf_mask is not None:
+            log_prob = log_prob * self.log_pdf_mask
+        return torch.sum(log_prob, -1).contiguous().view(batch_log_pdf_shape)
+
+
+class Sigmoid(Bijector):
```

Comment:
I'd rename this to `SigmoidBijector` and move it into `transformed_distributions.py`, since there are many other uses.

Reply:
+1 (that's what I figured, but didn't want to mess with `transformed_distributions` too much with approval)